### PR TITLE
Don't save the keychain before completing 2FA

### LINKED ISSUE #686: Exiting early on sign in with 2FA appear to be sign user in
### Description
If I begin to sign in with 2FA but then don't enter a 2FA code, I appear to be authenticated in Unity.

### Steps to Reproduce

1. Start signed out
2. Select sign in
3. Enter username and password
4. At the 2FA step, close the dialog

**Expected behavior:** 
To not be signed in

**Actual behavior:** 
I appear to be signed in:
screen shot 2018-04-10 at 10 42 33 am

`Go to profile` is successful

But then when I went to publish the repo I got this:
screen shot 2018-04-10 at 10 39 35 am

**Reproduces how often:** 
100%

### Additional Information

Logs that might be helpful:

```
180410-10:39:10.735 TRACE [104] <ProcessWrapper>                    Running '/Applications/Unity 2017.2.0f3/Unity.app/Contents/Tools/nodejs/bin/node "/Users/meaghanlewis/Library/Application Support/GitHubUnity/octorun/src/bin/app.js" login'
180410-10:39:11.683 TRACE [105] <ProcessWrapper>                    Running '/usr/local/bin/git credential-osxkeychain erase'
180410-10:39:12.140 TRACE [106] <ProcessWrapper>                    Running '/usr/local/bin/git credential-osxkeychain store'
180410-10:39:13.050 TRACE [ 1] <AuthenticationView>                Starting 2FA - Message:"Two Factor Required."
180410-10:39:25.263 TRACE [107] <ProcessWrapper>                    Running '/Applications/Unity 2017.2.0f3/Unity.app/Contents/Tools/nodejs/bin/node "/Users/meaghanlewis/Library/Application Support/GitHubUnity/octorun/src/bin/app.js" validate'
180410-10:39:25.507 TRACE [66] <RepositoryManager>                 WatcherOnRepositoryChanged
180410-10:39:25.570 TRACE [ 1] <GitStatusCache>                    Invalidate
180410-10:39:25.579 TRACE [108] <ProcessWrapper>                    Running '/usr/local/bin/git -c i18n.logoutputencoding=utf8 -c core.quotepath=false status -b -u --porcelain'
180410-10:39:25.668 TRACE [108] <RepositoryManager>                 IsBusyChanged Value:True
180410-10:39:25.760 TRACE [110] <RepositoryManager>                 IsBusyChanged Value:False
180410-10:39:25.874 TRACE [ 1] <GitStatusCache>                    04/10/2018 10:39:25 -07:00 Updating Entries: current:0 new:0
180410-10:39:25.881 TRACE [ 1] <GitAheadBehindCache>               04/10/2018 10:39:25 -07:00 Updating Ahead: current:0 new:0
180410-10:39:25.883 TRACE [ 1] <GitAheadBehindCache>               04/10/2018 10:39:25 -07:00 Updating Behind: current:0 new:0
180410-10:39:26.500 ERROR [ 1] <ApiClient>                         Error Getting Current User
GitHub.Unity.ApiClientException: Bad credentials
  at GitHub.Unity.ApiClient+<GetValidatedGitHubUser>c__asyncD.MoveNext () [0x00194] in C:\Users\Spade\Projects\GitHub\Unity\src\GitHub.Api\Application\ApiClient.cs:342 
=======
   at GitHub.Unity.ApiClient+<GetValidatedGitHubUser>c__asyncD.MoveNext() in C:\Users\Spade\Projects\GitHub\Unity\src\GitHub.Api\Application\ApiClient.cs:line 351
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.InvokeMoveNext(System.Object stateMachine)
   at System.Security.SecurityContext.Run(System.Security.SecurityContext securityContext, System.Threading.ContextCallback callback, System.Object state) in /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Security/SecurityContext.cs:line 150
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) in /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Threading/ExecutionContext.cs:line 150
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run()
   at System.Runtime.CompilerServices.TaskAwaiter+<>c.<OnCompletedInternal>b__17_1(System.Object state)
   at GitHub.Unity.MainThreadSynchronizationContext+<Post>c__AnonStorey1.<>m__0() in C:\Users\Spade\Projects\GitHub\Unity\src\UnityExtension\Assets\Editor\GitHub.Unity\Threading\SingleThreadSynchronizationContext.cs:line 20
```

### PR DESCRIPTION
**Pull request by StanleyGoldman**
Fixes #686 

- Don't save the keychain before completing 2FA
- Removing unused LoginAsync method

### CODE REVIEW

File: src/GitHub.Api/Authentication/LoginManager.cs [hunk 1]
```diff
@@ -106,7 +109,7 @@ class LoginManager : ILoginManager
             var host = loginResultData.Host;
             var keychainAdapter = keychain.Connect(host);
             var username = keychainAdapter.Credential.Username;
-            var password = keychainAdapter.Credential.Token;
```
Comment (shana) [inline src/GitHub.Api/Authentication/LoginManager.cs]: Is this change needed? The username and password are stored in the keychain at the beginning of the process (in https://github.com/github-for-unity/Unity/pull/691/files#diff-6d6481eca7dfd1acee9426efc4be8854R75), and we always use them until we're finished with 2fa, right?

Comment (StanleyGoldman) [reply src/GitHub.Api/Authentication/LoginManager.cs]: I have to test a bit more to properly address this question, but this change is necessary as a result of the changes above.

Without the change above the user interface would think the user is authenticated at this point. In order to prevent that I intentionally did not save the token in the keychain. Which is why I need to pull the token from the previous login result.

Comment (StanleyGoldman) [reply src/GitHub.Api/Authentication/LoginManager.cs]: Okay yea, we just need to not save the keychain.
